File: CS315-HW1/Codes/altiparmak_gokberk.dart
void main() {
  Map<String, String> capitals = {
    "Turkey" : "Ankara",
    "Italy" : "Rome"
};
  print(capitals);
  print(capitals["Turkey"]); // get element
  capitals["Germany"] = "Munich"; // add element
  print(capitals);
  capitals.remove("Italy"); // remove element
  print(capitals);
  capitals.update("Germany", (value) => "Berlin"); // update element
  print(capitals); 
  print(capitals.containsKey("Turkey")); // contains key
  print(capitals.containsValue("Ankara")); // contains value
  foo(capitals);
}
void foo(Map<String, String> aMap) {
  aMap.forEach((key, value) => print("key: $key value: $value")) ;
}

File: CS315-HW1/Codes/altiparmak_gokberk.js
var capitals = {
    "Turkey" : "Istanbul",
    "Spain" : "Madrid"
};
foo(capitals);
console.log("-----------");
console.log(capitals["Turkey"]); //get element
console.log("-----------");
capitals["Germany"] = "Berlin"; // add element
foo(capitals);
console.log("-----------");
delete capitals["Spain"]; // remove element
foo(capitals);
console.log("-----------");
capitals["Turkey"] = "Ankara";
foo(capitals);
console.log("-----------");
console.log("Turkey" in capitals); // contain key
console.log("-----------");
console.log(containsValue(capitals, "Ankara")); // contain value
console.log("-----------");
foo(capitals);

function containsValue(a, b){
    for (let key in a) {
	let value = a[key];
	if (value == b)
	    return true;
	return false;
    }
}

function foo(a) {
    for (let key in a){
	let value = a[key];
	console.log("Key: " + key, " Value: " + value);
    }
}
